PiggyBank/Domain/Models/DomainOperationModel.swift

```diff
@@ -3,53 +3,47 @@ import Foundation
 final class DomainOperationModel {
     
     enum OperationType: UInt {
-        case undefined = 0
+
         case budget = 1
         case transfer = 2
-        case plan = 3
+
     }
 
     let id: UInt
-    let categoryHexColor: String?
     let amount: Double
-    let accountTitle: String?
     let comment: String?
     let type: OperationType
-    let createdOn: Date
-    let planDate: Date?
-    let fromTitle: String?
-    let toTitle: String?
+    let date: Date
     let isDeleted: Bool
+    let category: DomainCategoryModel?
+    let fromAccount: DomainAccountModel
+    let toAccount: DomainAccountModel?
     
     init(
         id: UInt,
-        categoryHexColor: String?,
         amount: Double,
-        accountTitle: String?,
         comment: String?,
         type: UInt,
-        createdOn: String,
-        planDate: String?,
-        fromTitle: String?,
-        toTitle: String?,
+        date: String,
+        category: DomainCategoryModel?,
+        fromAccount: DomainAccountModel,
+        toAccount: DomainAccountModel?,
         isDeleted: Bool
     ) {
-        var planDateResult: Date?
-        if let plan = planDate, let date = plan.dateFromString() {
-            planDateResult = date
-        }
-
         self.id = id
-        self.categoryHexColor = categoryHexColor
-        self.amount = amount
-        self.accountTitle = accountTitle
         self.comment = comment
         self.type = OperationType(rawValue: type)!
-        self.createdOn = createdOn.dateFromString() ?? Date()
-        self.planDate = planDateResult
-        self.fromTitle = fromTitle
-        self.toTitle = toTitle
+        self.date = date.dateFromString() ?? Date()
+        self.category = category
+        self.fromAccount = fromAccount
+        self.toAccount = toAccount
         self.isDeleted = isDeleted
+        
+        var mutableAmount = amount
+        if let category = category, category.type == .outcome {
+            mutableAmount *= -1
```

Comment:
Better to do this into ViewModel or View

Reply:
I think this logic should be exactly here because this is common transformation among screens. If you want to switch back you can simple call `abs` or something like that. We can also make a separate function to return positive total always.

Reply:
I mean, model must not have any logic since it could potentially confuse during development because model need to store data not to modify

Reply:
The models which contains `as-is` data are `Data` layer objects. I mean they are network json models, db schemes etc. `Domain` layer models are much closer to business logic. Business logic may be covered by tests in future and it is more convenient not to write again and again logic behind total's sign there.

Reply:
I have implemented additional property 'absAmount' to return absolute value of operation's amount.